### PR TITLE
[EC-46] SFTP Connector

### PR DESCRIPTION
**Pull request by NileshN**
@amardaxini @yogendra689 @akshayEdCast 
Please refer code for SFTP connector. User input would be, 
- source machine IP address, 
- source machine host, 
- user to login to machine, 
- Base64 encoded password, 
- folder path in which CSV files present.

Used `Net::SFTP` class to make connection and fetch file content. 

Kindly review.

### CODE REVIEW

File: app/service/sftp_integration.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  def self.per_page
+    20
+  end
+
+  def get_content(options={})
+    decoded_pswd = Base64.decode64(@credentials['server_encrypted_password'])
+    sftp = Net::SFTP.start(@credentials['server_ip'], @credentials['server_username'], :password => decoded_pswd)
+    if sftp
+      sftp.dir.foreach(@credentials['server_folder_path']) do |entry|
```
Comment (yogendra689) [inline app/service/sftp_integration.rb]: not sure how many contents dir will have, can we check using findeach method?

File: app/service/sftp_integration.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        # ONLY CSV ALLOWED
+        if entry.file? && entry.name.include?("csv")
+          data = sftp.download!("#{@credentials['server_folder_path']}/#{entry.name}")
```
Comment (yogendra689) [inline app/service/sftp_integration.rb]: we already have csv_integration 
can you consider using csv_parser method.
Ref: https://github.com/Course-Master/rss_collector_app/blob/dev/app/service/read_csv_service.rb

let us know if you access or not to repo

Comment (NileshN) [reply app/service/sftp_integration.rb]: @yogendra689 Getting 404 error. Please provide access.

File: app/service/sftp_integration.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    20
+  end
+
+  def get_content(options={})
+    decoded_pswd = Base64.decode64(@credentials['server_encrypted_password'])
+    sftp = Net::SFTP.start(@credentials['server_ip'], @credentials['server_username'], :password => decoded_pswd)
+    if sftp
+      sftp.dir.foreach(@credentials['server_folder_path']) do |entry|
+        # ONLY CSV ALLOWED
+        if entry.file? && entry.name.include?("csv")
```
Comment (yogendra689) [inline app/service/sftp_integration.rb]: ends_with?('csv') can be used here

**Comment by NileshN** [comment]
@yogendra689 Please refer recent changes made to the connector. I have reused class created in EdCast's `rss_collector` repository.
Source object on the Integration framework:
`https://integrations-framework-qa.edcast.io/sources/8876878d-17f7-4e5d-8057-8bf0096f7f22`
CC: @akshayEdCast @darshan-patel

**Pull request merged**